src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxNodeTests.cs

```diff
@@ -2526,15 +2558,257 @@ class C
+        }
+
+        private void TestRemove_KeepEndOfLine(bool useOriginalEol)
+        {
+            var text = @"
+class C
+{
+// before
+void M()
+{
+} // after
+}";
+            var expectedText = @"
+class C
+{
+
+}";
+            if (!useOriginalEol)
+            {
+                text = text.NormalizeLineEndings();
+                expectedText = expectedText.NormalizeLineEndings();
+            }
+            var cu = SyntaxFactory.ParseCompilationUnit(text);
             var m = cu.DescendantNodes().OfType<MethodDeclarationSyntax>().FirstOrDefault();
             Assert.NotNull(m);
 
-            var cu2 = cu.RemoveNode(m, SyntaxRemoveOptions.KeepDirectives);
+            var removeOptions = SyntaxRemoveOptions.KeepEndOfLine;
+            if (useOriginalEol)
+            {
+                removeOptions |= SyntaxRemoveOptions.UseOriginalEndOfLine;
+            }
+            var cu2 = cu.RemoveNode(m, removeOptions);
+
+            text = cu2.ToFullString();
+
+            Assert.Equal(expectedText, text);
+        }
+
+        [Fact]
+        public void TestRemoveWithoutEOL_KeepEndOfLine()
+        {
+            TestRemoveWithoutEOL_KeepEndOfLine(false);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveWithoutEOL_KeepEndOfLine_UseOriginalEndOfLine()
+        {
+            TestRemoveWithoutEOL_KeepEndOfLine(true);
+        }
+
+        private void TestRemoveWithoutEOL_KeepEndOfLine(bool useOriginalEol)
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"class A { } class B { } // test");
+
+            var m = cu.DescendantNodes().OfType<TypeDeclarationSyntax>().LastOrDefault();
+            Assert.NotNull(m);
+
+            var removeOptions = SyntaxRemoveOptions.KeepEndOfLine;
+            if (useOriginalEol)
+            {
+                removeOptions |= SyntaxRemoveOptions.UseOriginalEndOfLine;
+            }
+            var cu2 = cu.RemoveNode(m, removeOptions);
 
             var text = cu2.ToFullString();
 
+            Assert.Equal("class A { } \r\n", text);
+        }
+
+        [Fact]
+        public void TestRemoveDocument_KeepEndOfLine()
+        {
+            TestRemoveDocument_KeepEndOfLine(false);
+        }
+
+        [Fact]
+        [WorkItem(22924, "https://github.com/dotnet/roslyn/issues/22924")]
+        public void TestRemoveDocument_KeepEndOfLine_UseOriginalEndOfLine()
+        {
+            TestRemoveDocument_KeepEndOfLine(true);
+        }
+
+        private void TestRemoveDocument_KeepEndOfLine(bool useOriginalEol)
+        {
+            var cu = SyntaxFactory.ParseCompilationUnit(@"
+#region A
+class A 
+{ } 
+#endregion");
+
+            var removeOptions = SyntaxRemoveOptions.KeepEndOfLine;
+            if (useOriginalEol)
+            {
+                removeOptions |= SyntaxRemoveOptions.UseOriginalEndOfLine;
+            }
+            var cu2 = cu.RemoveNode(cu, removeOptions);
```

Comment:
After merging the latest master, this line now throws a `NullReferenceException` on `Release`. After some investigation I found that [this PR](https://github.com/dotnet/roslyn/pull/22258) added a [AsRootOfNewTreeWithOptionsFrom](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs#L524) call that throws the exception. To me this looks like an unwanted breaking change introduced in `15.6`, or I am wrong?

Reply:
@maca88 It does sounds like a regression. Which part is `null` and causes the NRE?

Reply:
The [SyntaxNodeRemover.RemoveNodes call](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs#L524) in this case returns `null` which then causes a NRE as `AsRootOfNewTreeWithOptionsFrom` tries to [create a new tree](https://github.com/dotnet/roslyn/blob/74c7ec459a627a167914fbc796aea713481025a4/src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs#L432) with `null` as the root node.

Reply:
@maca88 Thanks for the clarification. It should be simple to include a check in `AsRootOfNewTreeWithOptionsFrom` to handle `null` input. Would you mind including that in your PR?
Thanks

Reply:
Included.